block/store.go

```diff
@@ -52,22 +53,46 @@ func StoreIncoming(packedBlock []byte, performRescan rescanType) error {
 	// extract incoming block record, checking for correct sequence
 	header, digest, data, err := blockrecord.ExtractHeader(packedBlock, height+1)
 	if nil != err {
+		globalData.log.Errorf("extract header error: %s", err)
 		return err
 	}
 
-	// ensure correct linkage
-	if previousBlock != header.PreviousBlock {
-		return fault.ErrPreviousBlockDigestDoesNotMatch
+	if blockrecord.IsDifficultyAppliedVersion(header.Version) {
+		if difficulty.IsAdjustBlock(header.Number) {
```

Comment:
I think you should make them in one line. 
`if blockrecord.IsDifficultyAppliedVersion(header.Version) && difficulty.IsAdjustBlock(header.Number)`

Reply:
yes, it should be in one line